src/components/dialog/dialog.component.tsx

```diff
@@ -91,6 +91,8 @@ export interface DialogProps extends ModalProps, TagProps {
   role?: string;
   /** Padding to be set on the Dialog content */
   contentPadding?: ContentPaddingInterface;
+  /** Allows developers to change the background color of the content to grey */
+  hasGreyBackground?: boolean;
```

Comment:
**nitpick(non-blocking)**: a minor suggestion to make the prop name and description more concise

```suggestion
  /** Change the background color of the content to grey */
  greyBackground?: boolean;
```